yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/commons/DockerPullImage.java

```diff
@@ -199,19 +198,19 @@ protected boolean shouldPullImage(DockerClient client, String imageName) {
             return false;
         }
 
-        List<Image> images = client.listImagesCmd().exec();
+        List<Image> images = client.listImagesCmd().withImageNameFilter(fullImageName).exec();
 
-        boolean hasImage = images.stream().anyMatch(image ->
-                nonNull(image.getRepoTags()) &&
-                        Arrays.stream(image.getRepoTags())
-                                .anyMatch(repoTag ->
-                                        repoTag.contains(fullImageName) || repoTag.contains("docker.io/" + fullImageName)
-                                )
-        );
+        boolean hasImage = false;
 
-        return hasImage ?
-                getPullStrategy().pullIfExists(imageName) :
-                getPullStrategy().pullIfNotExists(imageName);
+        for (Image image : images) {
+            for (String repoTag : image.getRepoTags()) {
+                if (repoTag.contains(fullImageName) || repoTag.contains("docker.io/" + fullImageName)) {
+                    hasImage = true;
+                }
+            }
+        }
+
+        return hasImage ? getPullStrategy().pullIfExists(imageName) : getPullStrategy().pullIfNotExists(imageName);
```

Comment:
so how this code now differs with previous? If there is no logical changes please remove change.